scripts/lib/CIME/XML/generic_xml.py

```diff
@@ -170,13 +166,16 @@ def name(self, node):
     def text(self, node):
         return node.xml_element.text
 
-    def add_child(self, node, root=None):
+    def add_child(self, node, root=None, position=0):
         """
         Add element node to self at root
         """
         expect(not self.locked and not self.read_only, "locked")
         root = root if root is not None else self.root
-        root.xml_element.append(node.xml_element)
+        if position > 0:
```

Comment:
0 is a valid position. I think we want position=None to be default and this if check should check against None.